src/store/modules/facility/actions.ts

```diff
@@ -146,6 +146,46 @@ const actions: ActionTree<FacilityState, RootState> = {
     commit(types.FACILITY_CURRENT_UPDATED, facility)
   },
 
+  async fetchCreatedFacility({ commit, state }, payload) {
+    const currentCreatedFacility = JSON.parse(JSON.stringify(state.createdFacility))
+    if (currentCreatedFacility.facilityId === payload.facilityId) {
+      return
+    }
+
+    emitter.emit("presentLoader");
+    const params = {
+      inputFields: {
+        facilityId: payload.facilityId
+      },
+      entityName: "FacilityAndProductStore",
+      noConditionFind: "Y",
+      distinct: "Y",
+      fieldList: ['facilityId', 'facilityName', 'facilityTypeId'],
+      viewSize: 1
+    }
+
+    let facility = {} as any;
+
+    try {
+      const resp = await FacilityService.fetchFacilities(params)
+
+      if(!hasError(resp)) {
+        facility = resp.data.docs[0]
+      } else {
+        throw resp.data
+      }
+    } catch(error) {
+      logger.error(error)
+    }
+
+    emitter.emit("dismissLoader");
+    commit(types.FACILITY_CREATED_UPDATED, facility);
```

Comment:
I think no need to maintain a separate state we can use the same state which is used to store/fetch selected facility on detail page?